packages/jest-circus/src/utils.ts

```diff
@@ -177,9 +179,17 @@ export const callAsyncCircusFn = (
     // soon as `done` called.
     if (fn.length) {
       let returnedValue: unknown = undefined;
+
       const done = (reason?: Error | string): void => {
+        if (testOrHook.seenDone) {
+          throw new Error('Expected done to be called once, but it was called multiple times.');
```

Comment:
```suggestion
          throw new ErrorWithStack(
            'Expected done to be called once, but it was called multiple times.',
            done,
          );
```

import `ErrorWithStack` from `jest-util`. That should clean up the stack trace a bit